libraries/src/Image/Image.php

```diff
@@ -1172,4 +1172,141 @@ public function setThumbnailGenerate($quality = true)
 	{
 		$this->generateBestQuality = (boolean) $quality;
 	}
+
+	/**
+	 * Method to generate different sized versions of current image.
+	 *
+	 * @param   mixed    $imageSizes      String or array of strings. Example: $imageSizes = array('1200x800','800x600');
+	 * @param   integer  $creationMethod  1-3 resize $scaleMethod | 4 create by cropping | 5 resize then crop
+	 *
+	 * @return  array
+	 *
+	 * @since   4.1.0
+	 * @throws  \LogicException
+	 * @throws  \InvalidArgumentException
+	 */
+	public function generateResponsiveImages($imageSizes, $creationMethod = self::SCALE_INSIDE)
```

Comment:
Why don't you use "generateThumbs" instead of copy pasting it and renaming some vars?

Reply:
"joomla.png" was added for just testing as I've specified in my last commit. It will be deleted in the next commit.